plugins/darktrace/icon_darktrace/util/api.py

```diff
@@ -50,8 +56,49 @@ def _get_signature(self, request: str, date: str) -> str:
             hashlib.sha1,
         ).hexdigest()
 
-    def _call_api(self, method: str, path: str, data: dict = None, params: dict = None) -> dict:
+    def _get_headers(self, endpoint: str) -> Dict[str, Any]:
+        """Returns the headers needed for all the Darktrace API requests
+
+        :param endpoint: Endpoint for headers to be generated
+        :type endpoint: str
+
+        :return: Dictionary contains headers needed for Darktrace API requests
+        :rtype: Dict[str, Any]
+        """
+
         date = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        headers = {
+            "DTAPI-Token": self.public_token,
+            "DTAPI-Date": date,
+            "DTAPI-Signature": self._get_signature(endpoint, date),
+        }
+        return headers
+
+    def _process_request_response(self, response: requests.Response) -> Dict[str, Any]:
+        """Returns the headers needed for all the Darktrace API requests
+
+        :param response: Response from requests.request
+        :type response: requests.Response
+
+        :return: Dict containing JSON API response
+        :rtype: Dict[str, Any]
+        """
+
+        if response.status_code == 400:
+            raise PluginException(preset=PluginException.Preset.BAD_REQUEST, data=response.text)
+        if response.status_code == 401:
+            raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD, data=response.text)
+        if response.status_code == 403:
+            raise PluginException(preset=PluginException.Preset.API_KEY, data=response.text)
+        if response.status_code == 404:
+            raise PluginException(preset=PluginException.Preset.NOT_FOUND, data=response.text)
+        if 400 < response.status_code < 500:
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response.text)
+        if 200 <= response.status_code < 300:
+            return response.json() if response.text else {}
+        raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response.text)
```

Comment:
May be unlikely but should we account for 300 redirect status codes separately from server error?

Reply:
Oh, yea. I will connect that with `UNKNOWN` preset then. Will that be ok? 😄

Reply:
Yeah I think that sounds good.`UKNOWN` might be better than `SERVER_ERROR`.